### PR TITLE
Suggest an extension to the abstract and concrete syntaxes

### PR DESCRIPTION
**Pull request by gkellogg**
… for directly representing `text direction` as a component of a `literal`.

### CODE REVIEW

**Review by domel — COMMENTED** [review]
What about `rdf:CompoundLiteral`? Is it still on the table?

**Comment by gkellogg** [comment]
> What about `rdf:CompoundLiteral`? Is it still on the table?

It’s still in the [document](https://github.com/w3c/rdf-star-wg/blob/main/docs/text-direction.md#the-rdfcompoundliteral-class), but I feel that both that and i18n are more historical notes at this time. I’m hoping the WG can decide on a direction so we can get on with it.

File: docs/text-direction.md [hunk 1]
```diff
@@ -98,6 +98,37 @@ A related solution would update the grammars to allow _both_ a language tag and
 ] .
 ```
 
+## Extend the Abstract Syntax
+
+Fully integrating the concept of text direction in RDF suggests adding it to the abstract syntax, and updating the relevant concrete syntaxes accordingly. This would allow a fourth element to be added to the definition of an [RDF literal](https://w3c.github.io/rdf-concepts/spec/#dfn-rdf-literal), in addition to `lexical form`, `datatype IRI`, and `language tag`:
+
+> A literal in an [RDF graph](https://w3c.github.io/rdf-concepts/spec/#dfn-rdf-graph) consists of two ***to four*** elements:
+> * ...
+> * ***if and only if the [datatype IRI](https://w3c.github.io/rdf-concepts/spec/#dfn-datatype-iri) is `http://www.w3.org/1999/02/22-rdf-syntax-ns#langString`, `text direction`, MUST BE empty or one or `ltr` or `rtl`***.
```
Comment (TallTed) [inline docs/text-direction.md:107]:
```suggestion
> * ***if and only if the [datatype IRI](https://w3c.github.io/rdf-concepts/spec/#dfn-datatype-iri) is `http://www.w3.org/1999/02/22-rdf-syntax-ns#langString`, `text direction`, MUST BE empty or one of `ltr` or `rtl`***.
```

File: docs/text-direction.md [hunk 2]
```diff
@@ -98,6 +98,37 @@ A related solution would update the grammars to allow _both_ a language tag and
...
+
+> A literal is a `language-tagged string` if the third element is present. Lexical representations of language tags MAY be converted to lower case. The value space of language tags is always in lower case. ***The fourth element, `text direction`, MUST NOT be present unless the `language tag` element is present. The value space of `text direction` is either empty, or one of `ltr` or `rtl`.
+
+> The `literal value` associated with a [literal](https://w3c.github.io/rdf-concepts/spec/#dfn-rdf-literal) is:
+> 1. If the literal is a [language-tagged string](https://w3c.github.io/rdf-concepts/spec/#dfn-language-tagged-string), then the literal value is a tuple consisting of its [lexical form](https://w3c.github.io/rdf-concepts/spec/#dfn-lexical-form) and its [language tag](https://w3c.github.io/rdf-concepts/spec/#dfn-language-tag), and its `text direction`, in that order.
```
Comment (TallTed) [inline docs/text-direction.md:112]:
```suggestion
> 1. If the literal is a [language-tagged string](https://w3c.github.io/rdf-concepts/spec/#dfn-language-tagged-string), then the literal value is a tuple consisting of its [lexical form](https://w3c.github.io/rdf-concepts/spec/#dfn-lexical-form), its [language tag](https://w3c.github.io/rdf-concepts/spec/#dfn-language-tag), and its `text direction`, in that order.
```

File: docs/text-direction.md [hunk 3]
```diff
@@ -98,6 +98,37 @@ A related solution would update the grammars to allow _both_ a language tag and
...
+> 2. ...
+
+> **Literal term equality**: Two literals are term-equal (the same RDF literal) if and only if the two [lexical forms](https://w3c.github.io/rdf-concepts/spec/#dfn-lexical-form), the two [datatype IRIs](https://w3c.github.io/rdf-concepts/spec/#dfn-datatype-iri), the two [language tags](https://w3c.github.io/rdf-concepts/spec/#dfn-language-tag) (if any), ***and the two `text directions`*** compare equal, character by character. Thus, two literals can have the same value without being the same RDF term.
+
+Plus, a paragraph expounding the the purpose of `text direction`.
+
+Concrete syntaxes, such as SPARQL and Turtle would change the definition of an `RDFLiteral`
```
Comment (TallTed) [inline docs/text-direction.md:119]:
```suggestion
Concrete syntaxes such as SPARQL and Turtle would change the definition of an `RDFLiteral`
```

**Review by TallTed — CHANGES_REQUESTED** [review]
Just a few tweaks

**Review by domel — APPROVED** [review]

**Pull request merged**